### ISSUE TITLE
Users updating packages can't understand output of pub

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/angular#1096]
Tracking issue for https://github.com/dart-lang/pub/issues/912.

Until this is complete, and rolled out, we can't release v5.0.0-final - our users have to go through mental gymnastics in order to figure out how to solve error messages like:

```
Resolving dependencies...
Package angular_components has no versions that match >=0.9.0-alpha+8 <0.10.0 derived from:
- merck_crm_mashup depends on version ^0.9.0-alpha+8
Process finished with exit code 1
```

**Comment by zoechi** [comment]
That's quite a common error message from `pub` - not only with Angular.
It also happened with every Angular release when used with angular_components when `pub upgrade` is run before the matching `angular_components` package was released which is usually a several days after an Angular release.

**Comment by eredo** [comment]
Isn't this fixed with 2.0.0-dev.44.0 and the new pub version resolver release?

**Comment by matanlurey** [comment]
Yup, happy to close this.
